### PR TITLE
Add ability to name, add, and delete schedules

### LINKED ISSUE #330: Improved Schedules
This is an epic to track progress of the "Improved Schedules" initiative.
This feature will also require changes to the backend.

Subtasks (in recommended order):
- [ ] #331
- [ ] #332
- [x] #333

Potential mockup from Design@UCI
image

Personally, I think it'd be better to have the edit icon on the far left and left align the text. Feel free to take liberties with the design!

### LINKED ISSUE #331: Add ability to rename schedules

### LINKED ISSUE #332: Add ability to create additional schedules
Users should be able to generate additional schedules. 

If possible, it'd be nice for them to reorder schedules in the dropdown list too.

### LINKED ISSUE #333: Start with one schedule, instead of 4 by default
Currently every user starts with 4 schedules by default. Change this behavior to start with just one and add additional schedules later.

### PR DESCRIPTION
**Pull request by Douglas-Hong**
## Summary
- Users can add any number of schedules they want with their own names. They can rename and delete these schedules.
- Users that already have saved schedules will start with 4 schedules (Schedule 1, Schedule 2, Schedule 3, Schedule 4).
- New users that do not have any saved schedules will start with just one schedule named `Schedule 1`.

## Test Plan
- Use the test plans from #350 and #368, and see if they pass.
- Make sure new users and existing users can properly save and load their schedules.

## Issues
- Closes #330 
- Closes #331 
- Closes #332 
- Closes #333

### CODE REVIEW

**Comment by Voark** [comment]
Every time a new schedule is added the text preview is "Schedule 2"
image
even when there are clearly more than 2 schedules. 

Also would it be worth it to populate with a default schedule name until a user clicks in the text box so someone can just create a new schedule without typing?

File: src/stores/AppStore.js [hunk 1]
```diff
@@ -204,14 +214,17 @@ class AppStore extends EventEmitter {
                 this.emit('colorChange', false);
                 break;
             case 'LOAD_SCHEDULE':
+                // If the user already had schedules, load up four schedules
```
Comment (Voark) [inline src/stores/AppStore.js]: I'm pretty sure I should've removed this comment when I made my changes but I forgot whoops.

**Review by Voark — APPROVED** [review]
Tentative approval as I don't know enough about components to review them but logic and functionality otherwise looks good.

**Review by ChaseC99 — APPROVED** [review]
Fantastic work on this feature 🎉

This was a huge task but you did an awesome job splitting it up into smaller PRs. After many code reviews and commits, this is ready to ship 🚢 

I’m so excited for students to use the new schedule system!

**Comment by ChaseC99** [comment]
Feel free to squash and merge when you're ready!

**Pull request merged**